primitive-types/impls/serde/src/serialize.rs

```diff
@@ -81,7 +81,7 @@ pub fn deserialize_check_len<'de, D>(deserializer: D, len: ExpectedLen) -> Resul
 		}
 
 		fn visit_str<E: de::Error>(self, v: &str) -> Result<Self::Value, E> {
-			if v.len() < 2 || &v[0..2] != "0x" {
+			if v.len() < 2 || !v.starts_with("0x") {
```

Comment:
This could be replaced `v.as_bytes()[0..2] != ['0' as u8, 'x' as u8]` if `starts_with` is too slow but I think it should be fast enough for such simple pattern

Reply:
Shouldn't we just  : 
```suggestion
			if !v.starts_with("0x") {
```